src/components/TransactionBlock/WalletTransactionBlock.tsx

```diff
@@ -0,0 +1,737 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { ethers } from 'ethers';
+import { AccountBalance, AccountTypes } from 'etherspot';
+
+import TextInput from '../TextInput';
+import { useEtherspot, useTransactionBuilder } from '../../hooks';
+import { formatAmountDisplay, formatAssetAmountInput, formatMaxAmount } from '../../utils/common';
+import { Chain, supportedChains } from '../../utils/chain';
+import NetworkAssetSelectInput from '../NetworkAssetSelectInput';
+import { IAssetWithBalance } from '../../providers/EtherspotContextProvider';
+import { CombinedRoundedImages, RoundedImage } from '../Image';
+import { Pill } from '../Text';
+import { Theme } from '../../utils/theme';
+import AccountSwitchInput from '../AccountSwitchInput';
+import { IMultiCallData, ITransactionBlock } from '../../types/transactionBlock';
+import {
+  WalletAssetSearchIcon,
+  WalletBridgeIcon,
+  WalletDepositIcon,
+  WalletSendIcon,
+  WalletSwapIcon,
+  WalletWithdrawIcon,
+  WalletDropdownUpIcon,
+  WalletDropdownDownIcon,
+  WalletCopyIcon,
+  WalletCloseSearchIcon,
+} from './Icons';
+import { Text } from '../Text';
+import { TRANSACTION_BLOCK_TYPE, TRANSACTION_BLOCK_TYPE_KEY } from '../../constants/transactionBuilderConstants';
+import { ISendAssetTransactionBlockValues } from './SendAssetTransactionBlock';
+
+interface IChainAssets {
+  title: string;
+  chainId: number;
+  chain: Chain;
+  assets: IAssetWithBalance[];
+}
+
+export interface IWalletTransactionBlock {
+  chain?: Chain;
+  availableTransactionBlocks: ITransactionBlock[];
+  hasTransactionBlockAdded: boolean;
+  addTransactionBlock: (block: ITransactionBlock, isBridgeDisabled: boolean, isKlimaIncluded: boolean) => void;
+}
+
+const WalletTransactionBlock = ({
+  chain,
+  availableTransactionBlocks,
+  hasTransactionBlockAdded,
+  addTransactionBlock,
+}: IWalletTransactionBlock) => {
+  const theme: Theme = useTheme();
+
+  const { setTransactionBlockValues, resetTransactionBlockFieldValidationError } = useTransactionBuilder();
+
+  const {
+    providerAddress,
+    accountAddress,
+    getSupportedAssetsWithBalancesForChainId,
+    smartWalletOnly,
+    smartWalletBalanceByChain,
+    sdk,
+  } = useEtherspot();
+
+  const [selectedNetwork, setSelectedNetwork] = useState<Chain | null>(supportedChains[1]);
```

Comment:
if Polygon chain is considered as default then would be good to make a lookup on `supportedChains` and find it by chain ID (137) rather then get by index number (think of order change for `supportedChains`)